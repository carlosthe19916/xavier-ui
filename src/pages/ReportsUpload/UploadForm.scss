.percentage-box {
    border-top: 0 !important;
    border-right: 0 !important;
    border-bottom: 0 !important;
    color: inherit !important;
}

.year-box {
    border-top: 0 !important;
    border-left: 0 !important;
    border-bottom: 0 !important;
    color: inherit !important;
}
