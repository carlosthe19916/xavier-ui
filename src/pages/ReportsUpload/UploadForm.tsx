import React from 'react';
import {
    FormikState,
    FormikValues,
    FormikActions,
    FormikHandlers
} from 'formik';
import {
    Form,
    FormGroup,
    InputGroup,
    TextInput,
    Button,
    ButtonVariant,
    Gallery,
    GalleryItem,
    InputGroupText,
    Stack,
    StackItem,
    ActionGroup
} from '@patternfly/react-core';
import Dropzone from 'react-dropzone';
import './UploadForm.scss';

interface UploadFormProps extends FormikState<FormikValues>, FormikActions<FormikValues>, FormikHandlers {
    file: File;
    onFileSelected: Function;
    handleCancel: (e: any) => any;
}

class UploadForm extends React.Component<UploadFormProps, { }> {

    constructor(props: UploadFormProps) {
        super(props);
    }

    componentDidMount() {
        this.updateFieldValues();
    }

    componentDidUpdate(prevProps: any) {
        if (prevProps.file !== this.props.file) {
            this.updateFieldValues();
        }
    }

    updateFieldValues() {
        if (this.props.file) {
            const fileName = this.props.file.name;
            setTimeout(() => {
                this.props.setFieldValue('file', fileName);
                if (!this.props.values.reportName) {
                    this.props.setFieldValue('reportName', fileName);
                }
            }, 0);
        }
    }

    render() {
        const {
            values,
            errors,
            touched,
            handleChange,
            handleBlur,
            handleSubmit,
            isSubmitting
        } = this.props;

        return (
            <Form onSubmit={ handleSubmit }>
                <FormGroup
                    isRequired
                    label="Inventory data file"
                    fieldId="file"
                    helperTextInvalid={ errors.file }
                    isValid={
                        (errors.file && touched.file) ? false : true
                    }
                >
                    <Dropzone
                        onDrop={ (files: File[]) => {
                            this.props.onFileSelected(files);
                        } }
                        noClick noKeyboard
                        multiple={ false }
                        accept={ [ 'application/zip', 'application/json' ] }
                    >
                        { ({ getRootProps, getInputProps, open }) => {
                            return (
                                <div className="container">
                                    <div { ...getRootProps({ className: 'dropzone' }) }>
                                        <input { ...getInputProps() } />
                                        <InputGroup>
                                            <TextInput
                                                id="file"
                                                name="file"
                                                type="text"
                                                aria-label="Select a File"
                                                value={ values.file }
                                                isValid={
                                                    (errors.file && touched.file) ? false : true
                                                } />
                                            <Button
                                                variant={ ButtonVariant.secondary }
                                                aria-label="Browse a file to upload"
                                                onClick={ open }
                                            >
                                                Browse
                                            </Button>
                                        </InputGroup>
                                    </div>
                                </div>
                            );
                        } }
                    </Dropzone>
                </FormGroup>
                <FormGroup
                    isRequired
                    label="Report name"
                    fieldId="reportName"
                    helperTextInvalid={ errors.reportName }
                    isValid={
                        (errors.reportName && touched.reportName) ? false : true
                    }
                >
                    <TextInput
                        isRequired
                        type="text"
                        id="reportName"
                        name="reportName"
                        aria-describedby="Report name"
                        onChange={ (_value, event) => handleChange(event) }
                        onBlur={ handleBlur }
                        value={ values.reportName }
                        isValid={
                            (errors.reportName && touched.reportName) ? false : true
                        }
                    />
                </FormGroup>
                <FormGroup
                    label="Report description"
                    fieldId="reportDescription"
                    helperTextInvalid={ errors.reportDescription }
                    isValid={ (errors.reportDescription && touched.reportDescription) ? false : true }
                >
                    <TextInput
                        type="text"
                        id="reportDescription"
                        name="reportDescription"
                        aria-describedby="Report description"
                        onChange={ (_value, event) => handleChange(event) }
                        onBlur={ handleBlur }
                        value={ values.reportDescription }
                        isValid={
                            (errors.reportDescription && touched.reportDescription) ? false : true
                        }
                    />
                </FormGroup>
                <FormGroup
                    isRequired
                    label="Year-over-year growth rate for new hypervisors"
                    fieldId="yearOverYearGrowthRatePercentage"
                    helperTextInvalid={ errors.yearOverYearGrowthRatePercentage }
                    isValid={
                        (errors.yearOverYearGrowthRatePercentage && touched.yearOverYearGrowthRatePercentage) ? false : true
                    }
                >
                    <Gallery>
                        <GalleryItem>
                            <InputGroup>
                                <TextInput
                                    isRequired
                                    id="yearOverYearGrowthRatePercentage"
                                    type="number"
                                    name="yearOverYearGrowthRatePercentage"
                                    aria-describedby="Year-over-year growth rate for new hypervisors"
                                    className="pf-u-text-align-right"
                                    onChange={ (_value, event) => handleChange(event) }
                                    onBlur={ handleBlur }
                                    value={ values.yearOverYearGrowthRatePercentage }
                                    isValid={
                                        (errors.yearOverYearGrowthRatePercentage && touched.yearOverYearGrowthRatePercentage) ? false : true
                                    }
                                />
                                <InputGroupText className="percentage-box">%</InputGroupText>
                            </InputGroup>
                        </GalleryItem>
                    </Gallery>
                </FormGroup>
                <Stack gutter="sm" className="pf-sm-gutter">
                    <StackItem isFilled={ false }>
                        <FormGroup
                            label="Percentage of hypervisors migrated each year"
                            fieldId="percentageOfHypervisorsMigratedOnYear1"
                            helperTextInvalid={ errors.percentageOfHypervisorsMigratedOnYear1 || errors.percentageOfHypervisorsMigratedSum }
                            isValid={
                                (errors.percentageOfHypervisorsMigratedOnYear1 && touched.percentageOfHypervisorsMigratedOnYear1) ||
                                (errors.percentageOfHypervisorsMigratedSum) ? false : true
                            }
                        >
                            <Gallery>
                                <GalleryItem>
                                    <InputGroup>
                                        <InputGroupText className="year-box">Year 1</InputGroupText>
                                        <TextInput
                                            isRequired
                                            id="percentageOfHypervisorsMigratedOnYear1"
                                            type="number"
                                            name="percentageOfHypervisorsMigratedOnYear1"
                                            aria-label="Percentage of hypervisors migrated on year 1"
                                            className="pf-u-text-align-right"
                                            onChange={ (_value, event) => handleChange(event) }
                                            onBlur={ handleBlur }
                                            value={ values.percentageOfHypervisorsMigratedOnYear1 }
                                            isValid={
                                                (errors.percentageOfHypervisorsMigratedOnYear1
                                                    && touched.percentageOfHypervisorsMigratedOnYear1) ? false : true
                                            }
                                        />
                                        <InputGroupText className="percentage-box">%</InputGroupText>
                                    </InputGroup>
                                </GalleryItem>
                            </Gallery>
                        </FormGroup>
                    </StackItem>
                    <StackItem isFilled={ false }>
                        <FormGroup
                            label=""
                            fieldId=""
                            helperTextInvalid={ errors.percentageOfHypervisorsMigratedOnYear2 || errors.percentageOfHypervisorsMigratedSum }
                            isValid={
                                (errors.percentageOfHypervisorsMigratedOnYear2 && touched.percentageOfHypervisorsMigratedOnYear2) ||
                                (errors.percentageOfHypervisorsMigratedSum) ? false : true
                            }
                        >
                            <Gallery>
                                <GalleryItem>
                                    <InputGroup>
                                        <InputGroupText className="year-box">Year 2</InputGroupText>
                                        <TextInput
                                            isRequired
                                            id="percentageOfHypervisorsMigratedOnYear2"
                                            type="number"
                                            name="percentageOfHypervisorsMigratedOnYear2"
                                            aria-label="Percentage of hypervisors migrated on year 2"
                                            className="pf-u-text-align-right"
                                            onChange={ (_value, event) => handleChange(event) }
                                            onBlur={ handleBlur }
                                            value={ values.percentageOfHypervisorsMigratedOnYear2 }
                                            isValid={
                                                (errors.percentageOfHypervisorsMigratedOnYear2
                                                    && touched.percentageOfHypervisorsMigratedOnYear2) ? false : true
                                            }
                                        />
                                        <InputGroupText className="percentage-box">%</InputGroupText>
                                    </InputGroup>
                                </GalleryItem>
                            </Gallery>
                        </FormGroup>
                    </StackItem>
                    <StackItem isFilled={ false }>
                        <FormGroup
                            label=""
                            fieldId=""
                            helperTextInvalid={ errors.percentageOfHypervisorsMigratedOnYear3 || errors.percentageOfHypervisorsMigratedSum }
                            isValid={
                                (errors.percentageOfHypervisorsMigratedOnYear3 && touched.percentageOfHypervisorsMigratedOnYear3) ||
                                (errors.percentageOfHypervisorsMigratedSum) ? false : true
                            }
                        >
                            <Gallery>
                                <GalleryItem>
                                    <InputGroup>
                                        <InputGroupText className="year-box">Year 3</InputGroupText>
                                        <TextInput
                                            isRequired
                                            id="percentageOfHypervisorsMigratedOnYear3"
                                            type="number"
                                            name="percentageOfHypervisorsMigratedOnYear3"
                                            aria-label="Percentage of hypervisors migrated on year 3"
                                            className="pf-u-text-align-right"
                                            onChange={ (_value, event) => handleChange(event) }
                                            onBlur={ handleBlur }
                                            value={ values.percentageOfHypervisorsMigratedOnYear3 }
                                            isValid={
                                                (errors.percentageOfHypervisorsMigratedOnYear3
                                                    && touched.percentageOfHypervisorsMigratedOnYear3) ? false : true
                                            }
                                        />
                                        <InputGroupText className="percentage-box">%</InputGroupText>
                                    </InputGroup>
                                </GalleryItem>
                            </Gallery>
                        </FormGroup>
                    </StackItem>
                </Stack>

                <ActionGroup>
                    <Button variant="primary" type="submit" isDisabled={ isSubmitting } disabled={ isSubmitting }>Create report</Button>
                    <Button variant="secondary" type="button" onClick={ this.props.handleCancel }>Cancel</Button>
                </ActionGroup>
            </Form>
        );
    }
}

export default UploadForm;
