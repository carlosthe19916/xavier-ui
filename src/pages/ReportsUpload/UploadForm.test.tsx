import { shallow } from "enzyme";
import UploadForm from "./UploadForm";

const props = {
  errors: {},
  values: {},
  touched: {},
  file: new File([ '' ], 'myFile.zip')
};

describe("UploadForm", () => {
  it("expect to render", () => {
    const wrapper = shallow(<UploadForm { ...props }/>);
    expect(wrapper).toMatchSnapshot();
  });

  it("expect to render errors", () => {
    const props1 = {
      ...props,
      errors: {
        file: 'Field value invalid',
        reportName: 'Field value invalid',
        reportDescription: 'Field value invalid',
        yearOverYearGrowthRatePercentage: 'Field value invalid',
        percentageOfHypervisorsMigratedOnYear1: 'Field value invalid',
        percentageOfHypervisorsMigratedOnYear2: 'Field value invalid',
        percentageOfHypervisorsMigratedOnYear3: 'Field value invalid',
        percentageOfHypervisorsMigratedSum: 'Field value invalid'
      },
      touched: {
        file: true,
        reportName: true,
        reportDescription: true,
        yearOverYearGrowthRatePercentage: true,
        percentageOfHypervisorsMigratedOnYear1: true,
        percentageOfHypervisorsMigratedOnYear2: true,
        percentageOfHypervisorsMigratedSum: true
      }
    };
    const wrapper = shallow(<UploadForm { ...props1 }/>);
    expect(wrapper).toMatchSnapshot();
  });
});
