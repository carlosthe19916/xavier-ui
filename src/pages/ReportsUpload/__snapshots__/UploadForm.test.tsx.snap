// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UploadForm expect to render 1`] = `
<Form
  className=""
  isHorizontal={false}
>
  <FormGroup
    className=""
    fieldId="file"
    isInline={false}
    isRequired={true}
    isValid={true}
    label="Inventory data file"
  >
    <Dropzone
      accept={
        Array [
          "application/zip",
          "application/json",
        ]
      }
      multiple={false}
      noClick={true}
      noKeyboard={true}
      onDrop={[Function]}
    >
      <Component />
    </Dropzone>
  </FormGroup>
  <FormGroup
    className=""
    fieldId="reportName"
    isInline={false}
    isRequired={true}
    isValid={true}
    label="Report name"
  >
    <TextInput
      aria-describedby="Report name"
      aria-label={null}
      className=""
      id="reportName"
      isDisabled={false}
      isReadOnly={false}
      isRequired={true}
      isValid={true}
      name="reportName"
      onChange={[Function]}
      type="text"
    />
  </FormGroup>
  <FormGroup
    className=""
    fieldId="reportDescription"
    isInline={false}
    isRequired={false}
    isValid={true}
    label="Report description"
  >
    <TextInput
      aria-describedby="Report description"
      aria-label={null}
      className=""
      id="reportDescription"
      isDisabled={false}
      isReadOnly={false}
      isRequired={false}
      isValid={true}
      name="reportDescription"
      onChange={[Function]}
      type="text"
    />
  </FormGroup>
  <FormGroup
    className=""
    fieldId="yearOverYearGrowthRatePercentage"
    isInline={false}
    isRequired={true}
    isValid={true}
    label="Year-over-year growth rate for new hypervisors"
  >
    <Gallery
      className=""
      gutter={null}
    >
      <GalleryItem>
        <InputGroup
          className=""
        >
          <TextInput
            aria-describedby="Year-over-year growth rate for new hypervisors"
            aria-label={null}
            className="pf-u-text-align-right"
            id="yearOverYearGrowthRatePercentage"
            isDisabled={false}
            isReadOnly={false}
            isRequired={true}
            isValid={true}
            name="yearOverYearGrowthRatePercentage"
            onChange={[Function]}
            type="number"
          />
          <InputGroupText
            className="percentage-box"
            component="span"
          >
            %
          </InputGroupText>
        </InputGroup>
      </GalleryItem>
    </Gallery>
  </FormGroup>
  <Stack
    className="pf-sm-gutter"
    component="div"
    gutter="sm"
  >
    <StackItem
      className=""
      isFilled={false}
    >
      <FormGroup
        className=""
        fieldId="percentageOfHypervisorsMigratedOnYear1"
        isInline={false}
        isRequired={false}
        isValid={true}
        label="Percentage of hypervisors migrated each year"
      >
        <Gallery
          className=""
          gutter={null}
        >
          <GalleryItem>
            <InputGroup
              className=""
            >
              <InputGroupText
                className="year-box"
                component="span"
              >
                Year 1
              </InputGroupText>
              <TextInput
                aria-label="Percentage of hypervisors migrated on year 1"
                className="pf-u-text-align-right"
                id="percentageOfHypervisorsMigratedOnYear1"
                isDisabled={false}
                isReadOnly={false}
                isRequired={true}
                isValid={true}
                name="percentageOfHypervisorsMigratedOnYear1"
                onChange={[Function]}
                type="number"
              />
              <InputGroupText
                className="percentage-box"
                component="span"
              >
                %
              </InputGroupText>
            </InputGroup>
          </GalleryItem>
        </Gallery>
      </FormGroup>
    </StackItem>
    <StackItem
      className=""
      isFilled={false}
    >
      <FormGroup
        className=""
        fieldId=""
        isInline={false}
        isRequired={false}
        isValid={true}
        label=""
      >
        <Gallery
          className=""
          gutter={null}
        >
          <GalleryItem>
            <InputGroup
              className=""
            >
              <InputGroupText
                className="year-box"
                component="span"
              >
                Year 2
              </InputGroupText>
              <TextInput
                aria-label="Percentage of hypervisors migrated on year 2"
                className="pf-u-text-align-right"
                id="percentageOfHypervisorsMigratedOnYear2"
                isDisabled={false}
                isReadOnly={false}
                isRequired={true}
                isValid={true}
                name="percentageOfHypervisorsMigratedOnYear2"
                onChange={[Function]}
                type="number"
              />
              <InputGroupText
                className="percentage-box"
                component="span"
              >
                %
              </InputGroupText>
            </InputGroup>
          </GalleryItem>
        </Gallery>
      </FormGroup>
    </StackItem>
    <StackItem
      className=""
      isFilled={false}
    >
      <FormGroup
        className=""
        fieldId=""
        isInline={false}
        isRequired={false}
        isValid={true}
        label=""
      >
        <Gallery
          className=""
          gutter={null}
        >
          <GalleryItem>
            <InputGroup
              className=""
            >
              <InputGroupText
                className="year-box"
                component="span"
              >
                Year 3
              </InputGroupText>
              <TextInput
                aria-label="Percentage of hypervisors migrated on year 3"
                className="pf-u-text-align-right"
                id="percentageOfHypervisorsMigratedOnYear3"
                isDisabled={false}
                isReadOnly={false}
                isRequired={true}
                isValid={true}
                name="percentageOfHypervisorsMigratedOnYear3"
                onChange={[Function]}
                type="number"
              />
              <InputGroupText
                className="percentage-box"
                component="span"
              >
                %
              </InputGroupText>
            </InputGroup>
          </GalleryItem>
        </Gallery>
      </FormGroup>
    </StackItem>
  </Stack>
  <ActionGroup
    className=""
  >
    <Button
      aria-label={null}
      className=""
      component="button"
      disabled={true}
      isActive={false}
      isBlock={false}
      isDisabled={true}
      isFocus={false}
      isHover={false}
      isInline={false}
      type="submit"
      variant="primary"
    >
      Create report
    </Button>
    <Button
      aria-label={null}
      className=""
      component="button"
      isActive={false}
      isBlock={false}
      isDisabled={false}
      isFocus={false}
      isHover={false}
      isInline={false}
      type="button"
      variant="secondary"
    >
      Cancel
    </Button>
  </ActionGroup>
</Form>
`;

exports[`UploadForm expect to render errors 1`] = `
<Form
  className=""
  isHorizontal={false}
>
  <FormGroup
    className=""
    fieldId="file"
    helperTextInvalid="Field value invalid"
    isInline={false}
    isRequired={true}
    isValid={false}
    label="Inventory data file"
  >
    <Dropzone
      accept={
        Array [
          "application/zip",
          "application/json",
        ]
      }
      multiple={false}
      noClick={true}
      noKeyboard={true}
      onDrop={[Function]}
    >
      <Component />
    </Dropzone>
  </FormGroup>
  <FormGroup
    className=""
    fieldId="reportName"
    helperTextInvalid="Field value invalid"
    isInline={false}
    isRequired={true}
    isValid={false}
    label="Report name"
  >
    <TextInput
      aria-describedby="Report name"
      aria-label={null}
      className=""
      id="reportName"
      isDisabled={false}
      isReadOnly={false}
      isRequired={true}
      isValid={false}
      name="reportName"
      onChange={[Function]}
      type="text"
    />
  </FormGroup>
  <FormGroup
    className=""
    fieldId="reportDescription"
    helperTextInvalid="Field value invalid"
    isInline={false}
    isRequired={false}
    isValid={false}
    label="Report description"
  >
    <TextInput
      aria-describedby="Report description"
      aria-label={null}
      className=""
      id="reportDescription"
      isDisabled={false}
      isReadOnly={false}
      isRequired={false}
      isValid={false}
      name="reportDescription"
      onChange={[Function]}
      type="text"
    />
  </FormGroup>
  <FormGroup
    className=""
    fieldId="yearOverYearGrowthRatePercentage"
    helperTextInvalid="Field value invalid"
    isInline={false}
    isRequired={true}
    isValid={false}
    label="Year-over-year growth rate for new hypervisors"
  >
    <Gallery
      className=""
      gutter={null}
    >
      <GalleryItem>
        <InputGroup
          className=""
        >
          <TextInput
            aria-describedby="Year-over-year growth rate for new hypervisors"
            aria-label={null}
            className="pf-u-text-align-right"
            id="yearOverYearGrowthRatePercentage"
            isDisabled={false}
            isReadOnly={false}
            isRequired={true}
            isValid={false}
            name="yearOverYearGrowthRatePercentage"
            onChange={[Function]}
            type="number"
          />
          <InputGroupText
            className="percentage-box"
            component="span"
          >
            %
          </InputGroupText>
        </InputGroup>
      </GalleryItem>
    </Gallery>
  </FormGroup>
  <Stack
    className="pf-sm-gutter"
    component="div"
    gutter="sm"
  >
    <StackItem
      className=""
      isFilled={false}
    >
      <FormGroup
        className=""
        fieldId="percentageOfHypervisorsMigratedOnYear1"
        helperTextInvalid="Field value invalid"
        isInline={false}
        isRequired={false}
        isValid={false}
        label="Percentage of hypervisors migrated each year"
      >
        <Gallery
          className=""
          gutter={null}
        >
          <GalleryItem>
            <InputGroup
              className=""
            >
              <InputGroupText
                className="year-box"
                component="span"
              >
                Year 1
              </InputGroupText>
              <TextInput
                aria-label="Percentage of hypervisors migrated on year 1"
                className="pf-u-text-align-right"
                id="percentageOfHypervisorsMigratedOnYear1"
                isDisabled={false}
                isReadOnly={false}
                isRequired={true}
                isValid={false}
                name="percentageOfHypervisorsMigratedOnYear1"
                onChange={[Function]}
                type="number"
              />
              <InputGroupText
                className="percentage-box"
                component="span"
              >
                %
              </InputGroupText>
            </InputGroup>
          </GalleryItem>
        </Gallery>
      </FormGroup>
    </StackItem>
    <StackItem
      className=""
      isFilled={false}
    >
      <FormGroup
        className=""
        fieldId=""
        helperTextInvalid="Field value invalid"
        isInline={false}
        isRequired={false}
        isValid={false}
        label=""
      >
        <Gallery
          className=""
          gutter={null}
        >
          <GalleryItem>
            <InputGroup
              className=""
            >
              <InputGroupText
                className="year-box"
                component="span"
              >
                Year 2
              </InputGroupText>
              <TextInput
                aria-label="Percentage of hypervisors migrated on year 2"
                className="pf-u-text-align-right"
                id="percentageOfHypervisorsMigratedOnYear2"
                isDisabled={false}
                isReadOnly={false}
                isRequired={true}
                isValid={false}
                name="percentageOfHypervisorsMigratedOnYear2"
                onChange={[Function]}
                type="number"
              />
              <InputGroupText
                className="percentage-box"
                component="span"
              >
                %
              </InputGroupText>
            </InputGroup>
          </GalleryItem>
        </Gallery>
      </FormGroup>
    </StackItem>
    <StackItem
      className=""
      isFilled={false}
    >
      <FormGroup
        className=""
        fieldId=""
        helperTextInvalid="Field value invalid"
        isInline={false}
        isRequired={false}
        isValid={false}
        label=""
      >
        <Gallery
          className=""
          gutter={null}
        >
          <GalleryItem>
            <InputGroup
              className=""
            >
              <InputGroupText
                className="year-box"
                component="span"
              >
                Year 3
              </InputGroupText>
              <TextInput
                aria-label="Percentage of hypervisors migrated on year 3"
                className="pf-u-text-align-right"
                id="percentageOfHypervisorsMigratedOnYear3"
                isDisabled={false}
                isReadOnly={false}
                isRequired={true}
                isValid={true}
                name="percentageOfHypervisorsMigratedOnYear3"
                onChange={[Function]}
                type="number"
              />
              <InputGroupText
                className="percentage-box"
                component="span"
              >
                %
              </InputGroupText>
            </InputGroup>
          </GalleryItem>
        </Gallery>
      </FormGroup>
    </StackItem>
  </Stack>
  <ActionGroup
    className=""
  >
    <Button
      aria-label={null}
      className=""
      component="button"
      disabled={true}
      isActive={false}
      isBlock={false}
      isDisabled={true}
      isFocus={false}
      isHover={false}
      isInline={false}
      type="submit"
      variant="primary"
    >
      Create report
    </Button>
    <Button
      aria-label={null}
      className=""
      component="button"
      isActive={false}
      isBlock={false}
      isDisabled={false}
      isFocus={false}
      isHover={false}
      isInline={false}
      type="button"
      variant="secondary"
    >
      Cancel
    </Button>
  </ActionGroup>
</Form>
`;
